javalin/src/main/java/io/javalin/router/Endpoint.kt

```diff
@@ -1,12 +1,33 @@
 package io.javalin.router
 
+import io.javalin.http.Context
 import io.javalin.http.Handler
 import io.javalin.http.HandlerType
 import io.javalin.security.RouteRole
+import io.javalin.util.mock.ContextMock
+import io.javalin.util.mock.Body
+import io.javalin.util.mock.MockConfig
+import java.util.function.Consumer
 
 class Endpoint @JvmOverloads constructor(
     val method: HandlerType,
     val path: String,
     val roles: Set<RouteRole> = emptySet(),
     val handler: Handler
-)
+) {
+
+    fun handle(ctx: Context): Context {
+        handler.handle(ctx)
+        return ctx
+    }
+
+    @JvmOverloads
+    fun handle(
+        ctx: ContextMock,
+        uri: String = path,
+        body: Body? = null,
+        cfg: (Consumer<MockConfig>)? = null
+    ): Context =
+        ctx.withMockConfig { cfg?.accept(it) }.execute(this, uri, body)
```

Comment:
If we want to follow proposed this order:

> val context = TestController.defaultApiEndpoint.createMockContext()

We need to mount this API at some point. It's purely in-memory impl that can be easily a part of the core implementation. I guess this is a convenient trade-off, because it's just a simple one-liner. Ofc we can abstract this behavior to something like:

```
TestController.defaultApiEndpoint.handle(mockCtx.build(uri, body) {
 // optional MockConfig cfg [...]
})
``` 

Or something like that, it'd be less elegant but `Endpoint` could stay quite generic.